AR_AttitudeControl: fix throttle jump on E-stop release

IamPete1 opened the pull request:
Fixes #16804

In order to stop @rmackay9 adding a extra param in #18606 ;)

We were ignoring the accel limit when the controller had not been active, this fixes that. And adds e-stop or being disarmed as things that mean we don't run the controller.

rmackay9 commented:
I think maybe we shouldn't add the dependency to SRV_Channel in the attitude controller but instead we should have an early-exit block at the top of each flight mode like we do in copter.

IamPete1 commented:
@rmackay9 The trouble with doing it in rover is that you have to remember to add that implementation, doing it directly where we apply the PID means that it is never bypassed or forgotten. Also means we only have to add it in one place of course. I don't think this is too offensive of a dependency.

IamPete1 commented:
@rmackay9 I have moved the Estop and armed check up into Rover. Also found and removed a pointless `stopped` bool we were passing all over the place and never using.

rmackay9 commented on libraries/APM_Control/AR_AttitudeControl.cpp:
```diff
@@ -802,9 +802,8 @@ float AR_AttitudeControl::get_desired_speed() const
 {
-    // return input value if no recent calls to speed controller
     // apply no limiting when ATC_ACCEL_MAX is set to zero
-    if (!speed_control_active() || !is_positive(_throttle_accel_max)) {
```
Have you tested what happens though the engaging the speed controller at a very different speed that it was previously active at?

For example, drive the vehicle at high speed in Manual mode and then switch to Acro... does the desired_speed start from zero or from the current speed?  Conversely drive the vehicle in Acro at high speed and then switch to manual, slow down and switch back to Acro.

rmackay9 commented:
Thanks again for this.

I'm very happy to merge these two commits:

- AR_AttitudeControl: get_throttle_out_stop: remove unused stopped bool
- Rover: remove unused stop bool

Just need to confirm with testing the behaviour of this commit:

- AR_AttitudeControl: fix throttle jump on E-stop release

.. but regarding this E-Stop handling, I'm still not a fan.  Like I said, "we should have an early-exit block at the top of each flight mode like we do in copter.".  The jump caused by E-stop is just one symptom of a larger problem which is that none of the flight modes are handling the disarmed (or E-stopped) state properly.  They're all working away as if they have control of the vehicle and this will lead to I-term build-up or other weirdness in some other scenario.

IamPete1 commented:
Was going to pull out those two into a separate PR, but there no-longer valid. Since https://github.com/ArduPilot/ardupilot/commit/a0ef7a1476e8632cbf8443d7ec564c4f217a5fb5 we use that stopped bool.

rmackay9 commented:
Actually I'm now less sure about my comment, "we should have an early-exit block at the top of each flight mode like we do in copter".  I do think we should have this but I think the E-stop handling needs to be in two places.  The motors library should be aware of it a bit like Copter's interlock... but then the attitude controllers also need to handle it somehow.. and I'm less sure where this should be.. maybe where you have it in this PR is ok.

IamPete1 commented:
Right, the problem with just early returning in the mode run function is that the rover can still navigate in E-stop. Sailboats work as normal. Steering vehicles still need to steer, either by rudder or Ackerman while there slowing down. 

A copter in E-stop losses both throttle and control. A rover in E-stop only looses throttle (except balance bots, and skid steer).
